netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -111,9 +116,22 @@ public NettyClientHandler(BufferingHttp2ConnectionEncoder encoder, Http2Connecti
     nextStreamId = connection.local().nextStreamId();
     connection.addListener(new Http2ConnectionAdapter() {
       @Override
+      public void onStreamClosed(Http2Stream stream) {
+        // Report the closure status for the stream.
+        reportStatus(clientStream(stream), null, false);
+      }
+
+      @Override
       public void onGoAwayReceived(int lastStreamId, long errorCode, ByteBuf debugData) {
         goAwayStatus(statusFromGoAway(errorCode, debugData));
```

Comment:
Yes this callback occurs before the streams are closed: https://github.com/netty/netty/blob/ed4a89082bb29b9e7d869c5d25d6b9ea8fc9d25b/codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java#L172